### ISSUE TITLE
Meta-Issue: Transmission Fixity

### ISSUE THREAD

**Issue by dannylamb** [Islandora/documentation#867]
We currently utilize transmission fixity from Drupal to Fedora, but don't support it on the initial upload into Drupal.  We need to be able to provide it in the REST api as well as allow users to provide their own checksum when uploading through a form.

**Comment by jonathangreen** [comment]
We also need some reasonable way in the UI to let the user know their fixity failed when adding objects. Both individual objects, and batches. 

Would this functionality rely on the microservice we are talking about in #847?

**Comment by dannylamb** [comment]
I don't think it would.  This is just to get it in before https://github.com/Islandora-CLAW/CLAW/issues/847's microservice can check it on demand.

**Comment by dannylamb** [comment]
We also shouldn't forget about covering the use case of ingesting using the Migrate framework, since it's our batch replacement.

**Comment by mjordan** [comment]
Related, we should be verifying checksums on migration (not just in general Migrate usage). In other words, get the checksum from Islandora 7.x, migrate the file, then confirm that the 7.x and CLAW checksums match. Report mismatches. Would only work on 7.x's where checksums were enabled. This use of transmission fixity is related to #917.

**Comment by ajs6f** [comment]
@mjordan Definitely that's something that should be available, but it should be config'able for those sites for which fixity is a service that comes from well beneath the repository layer. For sites like (like us 😁 ) recalculating checksums between two Islandora instances (of any version) is just extra grinding for no more trust.

**Comment by dannylamb** [comment]
Converting this to a meta issue to track development of a transmission fixity feature.  Please refer to this issue in any subsequent issue to link them.

To further clarify the scope of this ticket, we're only concerned with providing a checksum when you upload a file, so that it can be verified once deposited in the repository.

**Comment by Natkeeran** [comment]
Further discussion about providing a UI for file and media is found here: https://github.com/Islandora-CLAW/CLAW/issues/1102

**Comment by mjordan** [comment]
For those using Islandora Workbench, it now supports transmission fixity validation. Docs are at https://mjordan.github.io/islandora_workbench_docs/fixity/.
